### PR TITLE
Add OnRowClick to Table component

### PR DESCRIPTION
**Pull request by brooklynDev**
This adds a new Parameter to the Table component called OnRowClick that will trigger whenever a user clicks on any of the rows in the table.

[[中文版模板 / Chinese template](https://github.com/ant-design-blazor/ant-design-blazor/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/table/Table.razor [hunk 1]
```diff
@@ -159,7 +159,8 @@ RenderFragment<(Table<TItem> table, IEnumerable<TItem> showItems, int level)> bo
             currentRowData.HasChildren = hasChildren;
                 <CascadingValue Value="table.ColumnContext">
                     <CascadingValue Value="currentRowData" Name="RowData">
-                        <tr data-row-id="@rowIndex" class="ant-table-row ant-table-row-level-@level @(currentRowData.Selected ? "ant-table-row-selected" : "")">
+                        <tr @attributes="@(new Dictionary<string,object> { {"onclick", _callbackFactory.Create(table, () => table.RowClick(data)) } })" 
+                             data-row-id="@rowIndex" class="ant-table-row ant-table-row-level-@level @(currentRowData.Selected ? "ant-table-row-selected" : "")">
```
Comment (ElderJames) [inline components/table/Table.razor:163]: I think it is better to export the `RowData` for more information of the row.

**Review by ElderJames — CHANGES_REQUESTED** [review]
Thank you for contribution @dmoonfire .

File: components/table/Table.razor (hunk 1 above)
Comment (dmoonfire) [reply components/table/Table.razor:163]: Tagging @brooklynDev.

Comment (brooklynDev) [reply components/table/Table.razor:163]: Done!

**Review by ElderJames — APPROVED** [review]

**Pull request merged**